cuebot/src/main/java/com/imageworks/spcue/dispatcher/HostReportHandler.java

```diff
@@ -384,6 +386,27 @@ private void changeNimbyState(DispatchHost host, RenderHost rh) {
         }
     }
 
+    /**
+     * Changes the Lock state of the host. Looks at the number of locked cores and sets host to
+     * locked if all cores are locked.
+     *
+     * @param host DispatchHost
+     * @param renderHost RenderHost
+     */
+    private void changeLockState(DispatchHost host, CoreDetail coreInfo) {
+        if (host.lockState != LockState.LOCKED) {
```

Comment:
I think the logic here would be a little easier to follow if it were:

```
if (host.lockState == LockState.LOCKED) {
    ...
} else if (host.lockState == LockState.OPEN) {
    ...
}
```

But maybe there's a reason you've done it this way?

Reply:
Good call. I think I can clean this up to something along the lines of:

'''
If the host is locked:
    If locked cores are less than total cores:
        Unlock Host.
Otherwise if all cores are locked  but host is not locked:
    Lock Host.
'''